### PR TITLE
Implement LastLedgerSequence

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement `getLastLedgerSequence` for the  `XpringClient` which uses official rippled protocol buffers. 

This method is not hooked up. Test coverage will temporarily decrease. 

### Context of Change

rippled implemented protocol buffer support natively, which uses a new and incompatible set of protocol buffers. The implementation in rippled was completed in: https://github.com/ripple/rippled/pull/3159.

Protocol buffer upgrades are being done in piecemeal fashion for ease of review. 

*Status*

- [x] [Generate new versions of the protocol buffers](https://github.com/xpring-eng/Xpring-JS/pull/108)
- [x] [Migrate existing code to a `legacy` subdirectory and namespace](https://github.com/xpring-eng/Xpring-JS/pull/109)
- [ ] Create code that works with the new protocol buffers from the rippled team
   - [x] [Create a new `NetworkClient` with fakes](https://github.com/xpring-eng/Xpring-JS/pull/111)
   - [x] [Create a new `DefaultXpringClient` with tests](https://github.com/xpring-eng/Xpring-JS/pull/114/files)
   - [x] Read only calls for new `DefaultXpringClient` (In progress in this PR)
   - [ ] Write calls fro `DefaultXpringClient`
- [x] [Add a boolean parameter to switch between implementations](https://github.com/xpring-eng/Xpring-JS/pull/130)

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

Users who use the new protocol buffers will be able to check a transaction's status. 

## Test Plan

Unit tests on CI continue to pass.

### CODE REVIEW

File: src/default-xpring-client.ts [hunk 1]
```diff
@@ -163,7 +163,8 @@ class DefaultXpringClient implements XpringClientDecorator {
   }
 
   public async getLastValidatedLedgerSequence(): Promise<number> {
-    throw new Error(XpringClientErrorMessages.unimplemented)
+    const getFeeResponse = await this.getFee()
```
Comment (intelliot) [inline src/default-xpring-client.ts]: We need to get the fee in order to get the validated ledger sequence?

Comment (keefertaylor) [reply src/default-xpring-client.ts]: Yes. The `Fee` rpc happens to contain this information and we only implemented the minimal subset of these APIs in rippled to make this happen so this is what we're using for now.

File: src/default-xpring-client.ts [hunk 2]
```diff
@@ -177,11 +178,9 @@ class DefaultXpringClient implements XpringClientDecorator {
     return new GetTxResponseWrapper(getTxResponse)
   }
 
-  // TODO(keefertaylor): Add tests for this method once send is hooked up. 
-  private async getFee(): Promise<XRPDropsAmount> {
-    const getFeeRequest = new GetFeeRequest()
+  private async getMinimumFee(): Promise<XRPDropsAmount> {
```
Comment (hbergren) [inline src/default-xpring-client.ts]: is `getMinimumFee` used right now? Maybe `git` is just making the diff look weird.

Comment (keefertaylor) [reply src/default-xpring-client.ts]: Not yet. We're building toward #136 (implement send) which needs to get a fee and generate a last ledger sequence number (which requires the current ledger sequence number) in order to generate a transaction. 

In general, these helper methods are just mirroring the existing helper methods in `default-xpring-client`: https://github.com/xpring-eng/Xpring-JS/blob/master/src/legacy/legacy-default-xpring-client.ts#L200. There are differences in implementation in both that we're using a different set of protocol buffers and we implemented a slightly different subset of RPCs in rippled (they functionally serve the same set of data though).

**Comment by keefertaylor** [comment]
It's fine. I wish that GitHub had a way to distinguish a chain of PRs and link them together. 

I should also be providing this context in my descriptions and will do so in the future.

**Comment by keefertaylor** [comment]
We know that this is an outstanding task, and it's tracked here: https://app.asana.com/0/1143722888069470/1155647690941929

**Pull request merged**